### PR TITLE
Add static library target

### PR DESCRIPTION
**Pull request by nside**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,3 +57,8 @@ add_test(osmxTest osmxTest)
 install(TARGETS osmx DESTINATION bin)
 add_custom_target(archive COMMAND dist/archive.sh ${OSMX_VERSION} ${CMAKE_SYSTEM_NAME})
 add_dependencies(archive osmx)
+
+add_library(osmx-static STATIC src/storage.cpp src/expand.cpp src/extract.cpp src/update.cpp src/region.cpp)
```
Comment (bdon) [inline CMakeLists.txt:61]: any reason we shouldn't just do `add_library(osmx STATIC...` so that the output is libosmx.a instead of libosmx-static.a ?

**Comment by nside** [comment]
good call! I tried that but it was conflicting with the previous executable target

**Comment by nside** [comment]
```
CMake Error at CMakeLists.txt:61 (add_library):
  add_library cannot create target "osmx" because another target with the
  same name already exists.  The existing target is an executable created in
  source directory "/Users/dlaprise/src/OSMExpress".  See documentation for
  policy CMP0002 for more details.


CMake Error at CMakeLists.txt:62 (set_property):
  set_property could not find TARGET osmx-static.  Perhaps it has not yet
  been created.


-- Configuring incomplete, errors occurred!
```

**Comment by bdon** [comment]
Seems fine then, thank you!

**Pull request merged**